subsys/net/conn_mgr/conn_mgr_connectivity.c

```diff
@@ -77,6 +79,11 @@ int conn_mgr_if_disconnect(struct net_if *iface)
 out:
 	k_mutex_unlock(binding->mutex);
 
+	/* Inform conn_mgr that this is a hard-down scenario, since it won't know from the
```

Comment:
Why not? Plus I'm not really sure if a conscious disconnect from the application side (i. e. calling this function) should have a side effect of putting the interface down. From the description, I understood auto-down would only apply for the scenarios where connection attempt failed.

Reply:
Maybe the description needs to be updated, but my intention with the auto-down feature is that, if it is enabled, it should apply under ***any*** circumstance that does not trigger a re-connect.

Since the connectivity implementation is expected to **_not_** attempt to reconnect after an application-requested disconnect, auto-down should apply

Basically auto-down's function description is: "Take the iface down if we don't expect it to be attempt to connect any more on its own"

I'll need to wait until next week to adjust the comments here, but I think I also phrased this poorly. I meant that we cannot rely on the `if_down` event that results from calling `api->disconnect` so we instead directly call `conn_mgr_conn_handle_iface_hard_down`

It is named `hard_down` to distinguish it from a normal `if_down` event that might occur due to a temporary loss of connection, for instance in persistence mode.

Reply:
Ok, but indeed the descriptions could be updated to better reflect that.

Reply:
I've updated all the descriptions now -- please let me know if they seem clearer!